### PR TITLE
[onedpl] requires tbbmalloc

### PR DESCRIPTION
**Pull request by planetmarshall**
Specify library name and version:  **onedpl/20200330**

onedpl requires `tbb::malloc` which is not enabled by default. (actually an "OPTIONAL_COMPONENT" but these appear to be required by conan)

Also removed `settings` as this is a header-only library.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/onedpl/all/conanfile.py [hunk 1]
```diff
@@ -14,9 +14,12 @@ class OneDplConan(ConanFile):
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/oneapi-src/oneDPL"
     topics = ("stl", "parallelism")
-    settings = "os", "arch", "build_type", "compiler"
```
Comment (ericLemanissier) [inline recipes/onedpl/all/conanfile.py:17]: you have to keep this. `self.info.header_only()` is properly called below, so multiple packages are not created anyway

File: recipes/onedpl/all/conanfile.py [hunk 2]
```diff
@@ -54,12 +59,16 @@ def package(self):
...
-        self.cpp_info.components["_onedpl"].names["cmake_find_package"] = "ParallelSTL"
-        self.cpp_info.components["_onedpl"].names["cmake_find_package_multi"] = "ParallelSTL"
+        find_package_name = "ParallelSTL"
+        target_name = "pstl"
+        self.cpp_info.filenames["cmake_find_package"] = find_package_name
+        self.cpp_info.filenames["cmake_find_package_multi"] = find_package_name
+        self.cpp_info.names["cmake_find_package"] = target_name
+        self.cpp_info.names["cmake_find_package_multi"] = target_name
+        self.cpp_info.set_property("cmake_file_name", find_package_name)
+        self.cpp_info.set_property("cmake_target_name", target_name)
```
Comment (ericLemanissier) [inline recipes/onedpl/all/conanfile.py:69]: I think this property now stores the absolute target name

File: recipes/onedpl/all/conanfile.py [hunk 3]
```diff
@@ -26,8 +30,9 @@ def _source_subfolder(self):
         return "source_subfolder"
 
     def configure(self):
-        if self.settings.compiler.cppstd:
-            tools.check_min_cppstd(self, 11)
```
Comment (ericLemanissier) [inline recipes/onedpl/all/conanfile.py:30]: Why was this removed ?

File: recipes/onedpl/all/conanfile.py (hunk 2 above)
Comment (planetmarshall) [reply recipes/onedpl/all/conanfile.py]: fixed in latest push

**Comment by ericLemanissier** [comment]
Please don't use force push when there are reviews. it forces reviewers to review the full change, instead of seeing what's new in a new commit.

File: recipes/onedpl/all/conanfile.py (hunk 2 above)
Comment (planetmarshall) [reply recipes/onedpl/all/conanfile.py]: That works locally but appears to fail on CI - 
```
  Syntax error in cmake code at

    /home/conan/w/BuildSingleReference/conan-center-index/recipes/onedpl/all/test_package/build/2da8b83836e27aacd43916b404a0e9d98f21239f/ParallelSTLTarget-release.cmake:95

  when parsing string

    ${pstl::ParallelSTL_FRAMEWORKS_RELEASE}
```

**Comment by planetmarshall** [comment]
CI doesn't seem to like the absolute target name:
```
  Syntax error in cmake code at

    /home/conan/w/BuildSingleReference/conan-center-index/recipes/onedpl/all/test_package/build/2da8b83836e27aacd43916b404a0e9d98f21239f/ParallelSTLTarget-release.cmake:95

  when parsing string

    ${pstl::ParallelSTL_FRAMEWORKS_RELEASE}
```

Works locally though - CMake issue?

File: recipes/onedpl/all/conanfile.py [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 from conans import ConanFile, CMake, tools
 
-required_conan_version = ">=1.28.0"
+required_conan_version = ">=1.36.0"
```
Comment (ericLemanissier) [inline recipes/onedpl/all/conanfile.py:5]:
```suggestion
required_conan_version = ">=1.43.0"
```
The failure happens because the targets are are absolute since 1.43.0 (second item in https://docs.conan.io/en/latest/changelog.html#dec-2021)

**Comment by planetmarshall** [comment]
Giving this a nudge as I think the conan version has been updated now.

**Comment by ericLemanissier** [comment]
you can close/reopen the pull request to restart CI (but there seem to be some maintenance going on right now https://github.com/conan-io/conan-center-index/pull/8423#issuecomment-996563175)

File: recipes/onedpl/all/conanfile.py [hunk 5]
```diff
@@ -28,6 +32,9 @@ def _source_subfolder(self):
     def configure(self):
         if self.settings.compiler.cppstd:
             tools.check_min_cppstd(self, 11)
+        if self.options.backend == "tbb":
+            # Listed as an OPTIONAL_COMPONENT but optional components not supported by conan
+            self.options["tbb"].tbbmalloc = True
```
Comment (ericLemanissier) [inline recipes/onedpl/all/conanfile.py:37]: Please also raise an exception in validate if `self.options.backend == "tbb"` and `not self.options["tbb"].tbbmalloc`, because the option values set in `configure` can be changed by other recipes

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]
